contrib/golang/filters/http/test/test_data/basic/filter.go

```diff
@@ -162,10 +162,18 @@ func (f *filter) decodeHeaders(header api.RequestHeaderMap, endStream bool) api.
 	value := f.config.counter.Get()
 	header.Add("go-metric-counter-test-header-key", strconv.FormatUint(value, 10))
 
+	f.config.counter.Record(1)
```

Comment:
we'd better move these metric tests into another filter, i.e. metrics.go?

this might leads to test flake more frequency, it's just a guess, see this slack message for more details, https://envoyproxy.slack.com/archives/C04QNSXC7U0/p1692928578151599

Reply:
ok, i will try to move to another filter test.

Reply:
done, PTAL, thanks.